specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json

```diff
@@ -0,0 +1,416 @@
+              "$ref": "#/definitions/NatGateway"
+            },
+            "description": "Parameters supplied to the create or update nat gateway operation."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Create successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          },
+          "200": {
+            "description": "Update successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create nat gateway": { "$ref": "./examples/NatGatewayCreateOrUpdate.json" }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+           "final-state-via": "azure-async-operation"
+        }
+      },
+      "patch": {
+        "tags": [
+          "NatGateways"
+        ],
+        "operationId": "NatGateways_UpdateTags",
+        "description": "Updates nat gateway tags.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "natGatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the nat gateway."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./network.json#/definitions/TagsObject"
+            },
+            "description": "Parameters supplied to update nat gateway tags."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Update successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update nat gateway tags": { "$ref": "./examples/NatGatewayUpdateTags.json" }
+        },
+        "x-ms-long-running-operation": true
```

Comment:
Is this accurate? It looks like a synchronous operation based on responses and payload